docs/harbor-architecture.md

```diff
@@ -0,0 +1,47 @@
+# Harbor Architecture
+The following diagram[<sup>1</sup>](#edit-the-architecture-diagram) shows the various services that compose Harbor and how they are related.
+
+![rbac](img/harbor_architecture.png)
+
+## Services
+These are the responsibilities of the various services that compose Harbor.
+
+* nginx
+    * Reverse proxy for the ui and registry
+    * Terminates TLS (if used)
+* registry
+    * The [Docker Registry](https://docs.docker.com/registry/) is a stateless, highly scalable server side application that stores and lets you distribute Docker images
+    * Sends [notifications](https://docs.docker.com/registry/notifications/) to the ui
+* ui
+    * Serves the user inteface code to browsers
+    * Reverse proxy for registry API
+    * Sends jobs to job service
+    * Retrieves image vulnerability data from clair/postgres
+    * Retrieves configuration from adminserver
+    * Stores data in mysql
+* adminserver
+    * Authenticates users for ui and registry
```

Comment:
Does the adminserver actually handle auth?

It's not clear to me yet because there are auth packages at both [src/adminserver/auth](https://github.com/vmware/harbor/tree/master/src/adminserver/auth) and [src/ui/auth](https://github.com/vmware/harbor/tree/master/src/ui/auth).